File: src/CorporateMessengerWithAgile/Application/Dto/TeamDto.cs
namespace Application.Dto
{
    public class TeamDto : BaseDto
    {
        public Guid ProjectId { get; set; }
        public string Title { get; set; } = null!;
        public int StandardSprintDuration { get; set; }
        public IReadOnlyList<Guid> TeamMemberIds { get; set; } = null!;
        public IReadOnlyList<Guid> SprintIds { get; set; } = null!;
        public IReadOnlyList<Guid> KanbanBoardColumnIds { get; set; } = null!;
    }
}

File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/TeamDto.ts
// src/models/entity/TeamDto.ts
import { z } from 'zod';
import { BaseDtoSchema } from './BaseDto';
import { GuidSchema } from '../Guid';

export const TeamDtoSchema = BaseDtoSchema.extend({
    projectId: GuidSchema,
    title: z.string().min(1),
    standardSprintDuration: z.number(),
    teamMemberIds: z.array(GuidSchema).optional(),
    sprintIds: z.array(GuidSchema).optional(),
    kanbanBoardColumnIds: z.array(GuidSchema).optional(),
});

export type TeamDto = z.infer<typeof TeamDtoSchema>;
